atomate/atat/fireworks/core.py

```diff
@@ -0,0 +1,193 @@
+            }
+        else:
+            for cluster in clusters.keys():
+                if cluster not in [2, 3, 4, 5, 6]:
+                    raise ValueError("Mcsqs only supports clusters of 2-6 atoms.")
+
+        rndstr = disordered_struct.to(fmt='mcsqs')
+
+        cluster_str = " ".join(["{}={}".format(k, v) for k, v in clusters.items()])
+        generate_cluster_cmd = "mcsqs {}".format(cluster_str)
+
+        if user_input_settings:
+            user_input_settings_str = " ".join(["{}={}".format(k, v) for k, v
+                                            in user_input_settings.items()])
+        else:
+            user_input_settings_str = ""
+
+        # using same approach as Custodian to detect ncores
+        ncores = os.environ.get('NSLOTS') or multiprocessing.cpu_count()
+        ncores = int(ncores)
+
+        # mcsqs is not parallelised, and is not expected to finish within the walltime
+        # it's a Monte Carlo process that can run indefinitely, so we use timeout
+        # to ensure the command finishes
+        timeout_cmd = which('timeout') or which('gtimeout')
+        if not timeout_cmd:
+            raise RuntimeError("Requires timeout to limit walltime.")
+
+        with open(os.path.join(module_dir, 'run_mcsqs_template.sh')) as f:
+            template = f.read()
+
+        bash_script = Template(template)
+        bash_script = bash_script.substitute({
+            'timeout_cmd': timeout_cmd,
+            'time': 60*(walltime - 1),  # seconds, minus 60 seconds for DB task
+            'ncores': ncores,
+            'size': size,
+            'settings': user_input_settings_str
+        })
+        bash_script = str(bash_script)
+
+        # write our input args, so that they can be picked up by the drone
+        mcsqs_input_args = json.dumps({
+            'clusters': clusters,
+            'user_input_settings': user_input_settings,
+            'walltime': ncores*(walltime - 1)
+        })
+
+        files_to_write = [
+            {
+                'filename': 'rndstr.in',
+                'contents': rndstr
+            },
+            {
+                'filename': 'mcsqs_input_args.json',
+                'contents': mcsqs_input_args
+            }
+        ]
+
+        tasks = [
+            FileWriteTask(files_to_write=files_to_write),
+            ScriptTask(script=[
+                generate_cluster_cmd,
+                bash_script
+            ], shell_exe='/bin/bash')
+        ]
+
+        if db_file:
+            tasks.append(McsqsToDbTask(db_file=db_file))
+
+        super(McsqsFW, self).__init__(tasks, name=name, **kwargs)
+
+    @staticmethod
+    def _determine_min_cell(disordered_struct):
+        """
+        Get a minimum cell size for an ordered structure.
+        """
+        denoms = {}
+        for sp_occu in disordered_struct.species_and_occu:
+            for sp, occu in sp_occu.items():
+                denom = Fraction(occu).limit_denominator(100).denominator
```

Comment:
not sure why this is hard-coded as 100 rather than using the max_denominator set earlier in the FW

Reply:
Yeah I'm not sure why either. This function should really go in pymatgen somewhere anyway (e.g. as a method on disordered structures generally).

Reply:
Isn't this already the case? We already coded a proper disordered occupancy rounding in pymatgen.

Reply:
Yeah, there's DiscretizeOccupancyTransformation there now, but I meant the method for determining the smallest ordered cell size for a disordered structure -- I don't think that's in pymatgen?